sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -359,20 +386,129 @@
      */
     public Read<K, V> withTopicPartitions(List<TopicPartition> topicPartitions) {
-      checkState(getTopics().isEmpty(), "Only topics or topicPartitions can be set, not both");
-      return toBuilder().setTopicPartitions(ImmutableList.copyOf(topicPartitions)).build();
+      return withTopicPartitions(
+          ValueProvider.StaticValueProvider.of(Joiner.on(',').join(topicPartitions)));
+    }
+
+    /**
+     * Sets a list of partitions to read from. Partitions are provided as a comma separated list of
+     * Strings in the format: topic1-partition1,topic1-partition2...
+     * This allows reading only a subset of partitions for one or more topics when (if ever) needed.
+     *
+     * <p>See {@link KafkaUnboundedSource#split(int, PipelineOptions)} for description
+     * of how the partitions are distributed among the splits.
+     */
+    public Read<K, V> withTopicPartitions(String topicPartitions) {
+      return withTopicPartitions(ValueProvider.StaticValueProvider.of(topicPartitions));
+    }
+
+    /**
+     * Like above but with a {@link ValueProvider ValueProvider&lt;String&gt;}.
+     */
+    public Read<K, V> withTopicPartitions(ValueProvider<String> topicPartitions) {
+      checkState(getTopics().get().isEmpty(),
+          "Only topics or topicPartitions can be set, not both");
+      return toBuilder().setTopicPartitions(ValueProvider
+              .NestedValueProvider.of(topicPartitions, new TopicPartitionTranslator())).build();
+    }
+
+    /**
+     * Used to build a {@link ValueProvider} for {@link List List&lt;String&gt;}.
+     */
+    private static class TopicTranslator implements SerializableFunction<String, List<String>> {
+      @Override
+      public List<String> apply(String topics) {
+        return ImmutableList.copyOf(
+            Splitter.on(',').trimResults().omitEmptyStrings().splitToList(topics));
+      }
+
+    }
+
+    /**
+     * Used to build a {@link ValueProvider} for {@link List List&lt;TopicPartition&gt;}.
+     */
+    private static class TopicPartitionTranslator
+        implements SerializableFunction<String, List<TopicPartition>> {
+      @Override
+      public List<TopicPartition> apply(String topicPartitions) {
+        List<TopicPartition> topicPartitionList = new ArrayList<>();
+        for (String topicPartition: Splitter.on(',').trimResults().omitEmptyStrings()
+              .splitToList(topicPartitions)) {
+          topicPartitionList
+              .add(new TopicPartition(Splitter.on('-').splitToList(topicPartition).get(0),
+                  Integer.parseInt(Splitter.on('-').splitToList(topicPartition).get(1))));
+        }
+      return ImmutableList.copyOf(topicPartitionList);
+      }
+    }
+
+    /**
+     * Sets a Kafka {@link Deserializer Deserializer&lt;K&gt;} for interpreting key bytes read.
+     * This uses the {@link String} provided to set the Deserializer
+     */
+    public Read<K, V> withKeyDeserializer(String keyDeserializer) {
```

Comment:
Rename this to something like `withKeyDeserializerClassName`?

Reply:
This is used outside of KafkaIO. Not sure if renaming this may cause issues

Reply:
I have renamed withKeyDeserializerClass to withKeyDeserializerClassName